### ISSUE TITLE
element with id `undefined` is considered as saveBtn

### ISSUE THREAD

**Issue by nikhilmandlik** [devforth/painterro#129]
### **Describe the bug**
In [OpenMCT](https://github.com/nasa/openmct) we are using 'painterro' with 
`hiddenTools: ['save', 'open', 'close', 'eraser', 'pixelize', 'rotate', 'settings', 'resize'],`

One of the element has a id as `undefined` (clearly it is a bug in OpenMCT)

But with above edge case `saveBtn` is now same element with id =`undefined` and is getting attribute `disabled` `true`
```
this.saveBtn = this.doc.getElementById(this.toolByName.save.buttonId),
this.saveBtn && this.saveBtn.setAttribute("disabled", "true"),
```
may be need to check for `this.toolByName.save.buttonId === undefined` before `getElementById`?

**Painterro version**
"painterro": "^1.0.35",

###  **To Reproduce**
1. Create `painterro`  with `hiddenTools: ['save', 'open', 'close', 'eraser', 'pixelize', 'rotate', 'settings', 'resize'],`
2. have a element with id=`undefined`
3. now save painterro using `.save();`
4. observe element with id=`undefined` is now `disabled` 😅 

### **Screenshots**
<img width="1678" alt="Screen Shot 2021-04-06 at 3 05 54 PM" src="https://user-images.githubusercontent.com/1292612/113784123-b8636480-96e9-11eb-8ce3-9e56dab40eac.png">

<img width="1677" alt="Screen Shot 2021-04-06 at 3 06 40 PM" src="https://user-images.githubusercontent.com/1292612/113784137-bc8f8200-96e9-11eb-8fb4-32065c776f7d.png">

### **Browser**
`Google Chrome Version 89.0.4389.114 (Official Build) (x86_64)` on Mac

**Comment by ivictbor** [comment]
@nikhilmandlik Hello
And thanks for your issue. We are working on your suggestion.


BTW, did not you try to set a DOM breakpoint on ID attribute modification to understand which code in OpenMCT touches it? 

![](https://developers.google.com/web/updates/images/2015-05-20-view-and-change-your-dom-breakpoints-with-the-breakpoints-pane/dom-breakpoint-pane.gif)

I expected such issues, that's why I generated IDs in the format "ptro<random string>" for example "ptroQfBt5zKcYdMqmhwprTxz",
I am afraid that some selector in OpenMCT might accidentally found some Painterro element, and if this is true then I will need to find some "popular" class names in the Ptro and add special unique prefixes, but for me, it seems like each one should have it. 

if the branch you are working on is public, and is clear how to deploy and reproduce then we can take a look.

**Comment by Darktimeoff** [comment]
Hi @nikhilmandlik !
We found and fixed an issue in the latest release, it was only a Painterro bug.
Thing is that `this.toolByName.save.buttonId` is always undefined when the save tool is hidden,
So your fix is right. Plus I fixed access to the whole document at all, we should always do such actions only in Painterro container.

**Comment by ivictbor** [comment]
@nikhilmandlik please update to `painterro@1.2.37`, if you have any issues or questions please let us know any time and we will try address them ASAP

**Comment by nikhilmandlik** [comment]
Hello @Darktimeoff , @ivictbor 
Thanks a lot for quick reply 😃 

> BTW, did not you try to set a DOM breakpoint on ID attribute modification to understand which code in OpenMCT touches it?

@ivictbor yes I tried DOM breakpoint and was pointing to Painterro, and thought it was odd :P

Yes it was bug in both OpenMCT and Painterro 😅.
I guess thanks to bug in Painterro, was able to find OpenMCT bug .....lol.

Also wanted to report this so that Painterro covers that edge case. ✌️

**Comment by ivictbor** [comment]
@nikhilmandlik FYI found and fixed a very related bug (1.2.44 has all fixes), plus added a hook which would not allow to touch non painterro dom
